File: WebApplication1/WebApplication1/ClientApp/src/components/Order.js
import React, { Component } from 'react';
import { OrderForm } from './OrderForm'
import { OrdersList } from './OrdersList'

export class Order extends React.Component {
    constructor(props) {
        super(props);
        this.state = { data: props.order };
        this.onClick = this.onClick.bind(this);
    }
    onClick(e) {
        this.props.onRemove(this.state.data);
    }
    render() {
        return <div class='order'>
            <p>ID(номер) заказа: 
                <label>{this.state.data.id}</label>
            </p>
            <p>Город отправителя:
                <label>{this.state.data.senderCity}</label>
            </p>
            <p>Адрес отправителя:
                <label>{this.state.data.senderAddress}</label>
            </p>
            <p>Город получателя:
                <label>{this.state.data.recipientCity}</label>
            </p>
            <p>Адрес получателя:
                <label>{this.state.data.recipientAddress}</label>
            </p>
            <p>Вес груза:
                <label>{this.state.data.weight}</label>
            </p>
            <p>Дата забора груза:
                <label>{this.state.data.pickUpDate.substring(0, 10)}</label>
            </p>

            <input onClick={this.onClick} type="submit" value="Удалить" />
        </div>;
    }
}
export default Order

File: WebApplication1/Models/Models/Order.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Models.Models
{
    /// <summary>
    /// Класс, описывающий заказ
    /// </summary>
    public class Order
    {
        /// <summary>
        /// Уникальный идентификатор заказа.
        /// </summary>
        public Guid Id { get; set; }
        /// <summary>
        /// Город отправителя
        /// </summary>
        public string SenderCity { get; set; }
        /// <summary>
        /// Адрес отправителя
        /// </summary>
        public string SenderAddress { get; set; }
        /// <summary> 
        /// Город получателя
        /// </summary>
        public string RecipientCity { get; set; }
        /// <summary>
        /// Адрес получателя
        /// </summary>
        public string RecipientAddress { get; set; }
        /// <summary>
        /// Вес груза
        /// </summary>
        public decimal Weight { get; set; }
        /// <summary>
        /// Дата забора груза
        /// </summary>
        public DateTime PickUpDate { get; set; }
    }
}
